k2/csrc/properties.cc

```diff
@@ -50,4 +86,28 @@ bool IsEpsilonFree(const Fsa &fsa) {
   return true;
 }
 
+bool IsConnected(const Fsa &fsa) {
+  StateId num_states = fsa.NumStates();
+  std::vector<bool> accessible(num_states, false);
+  accessible[0] = true;  // the start state
+  for (const auto &arc : fsa.arcs) {
+    if (!accessible[arc.src_state]) return false;
+    accessible[arc.dest_state] = true;
+  }
+  if (std::find(accessible.begin(), accessible.end(), false) !=
+      accessible.end())
+    return false;
+
+  // reuse `accessible` to process co-accessible situation.
```

Comment:
```cpp
auto &coaccessible = accesible;
```
Give it an alias makes the code better to read.

Reply:
Actually, I plan to call your implementation of `ConnectCore` here as which will get less code totally. so just ignore this here.